### PR TITLE
[C++][Pistache]Fix import of model header

### LINKED ISSUE #2024: [BUG] [C++][Pistache-server] Object defined inside the body content does not compile
##### Description
When the YAML file describes the use of an object inside the body content (without any content type like application/json), the generator produces a model file called InlineObject.h (and cpp) but to use it, the generator set the wrong include file name :

```
#include "Inline_object.h"
```
instead of

```
#include "InlineObject.h"
```

It seems that there is a confusion between the file name and the class name

```
class  Inline_object
    : public ModelBase
{
    ...
} ;
```

##### openapi-generator version

Current master 4.0.0-SNAPSHOT

##### OpenAPI declaration file content or url

```yaml
openapi: 3.0.0
info:
  version: 1.0.0
  title: Check generation of pistache
  description: Internal ref filename is check_object_immediately_in_the_content.yaml

servers:
  - url: http://localhost:8080

paths:
  /sm-contexts:
    post:
      summary:  Create SM Context
      tags:
        - SM contexts collection
      operationId: PostSmContexts
      requestBody:
        description: representation of the SM context to be created in the SMF
        required: true
        content:
          multipart/related:
            schema:
              type: object
              properties: # Request parts
                dataMessage:
                  type: string

      responses:
        '200':
          description: Everythings gonna be alright
          content:
            application/json:
              schema:
                $ref: "#/components/schemas/Content"
        default:
          description: unexpected error


components:
  schemas:
    Content:
      type: string

```

##### Command line used for generation

Generate :
```
openapi-generator-cli.sh generate -i ./openapi.yaml -g cpp-pistache-server -c ./config.json -o .
```

Compile :
```
g++ -c  -I./api -I./model -I./impl  -Wall -g -std=c++11 -o obj/api/SMContextsCollectionApi.o api/SMContextsCollectionApi.cpp
```

##### Steps to reproduce

Generate & compile

##### Related issues/PRs

N/A

### PR DESCRIPTION
**Pull request by etherealjoy**
### PR checklist

- [X] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [X] Ran the shell script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run `./bin/{LANG}-petstore.sh`, `./bin/openapi3/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates). Windows batch files can be found in `.\bin\windows\`.
- [X] Filed the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`~~, `3.4.x`, `4.0.x`~~. Default: `master`.
- [X] Copied the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

### Description of the PR
Fixes #2024
- Fixes import of model header due to rework (similar to #2168, #2750, #2744) 
- sanitizing the name is done in `AbstractCppCodegen` already so not needed twice

@stkrwork @ravinikam @MartinDelille @fvarose

### CODE REVIEW

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/CppPistacheServerCodegen.java [hunk 1]
```diff
@@ -285,7 +285,7 @@ public String apiFilename(String templateName, String tag) {
 
     @Override
     public String toApiFilename(String name) {
-        return  modelNamePrefix + camelize(name) + "Api";
+        return  modelNamePrefix + toModelName(name) + "Api";
```
Comment (stkrwork) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/CppPistacheServerCodegen.java]: We use toModelName in toApiFilename? Wouldn't a toApiName make more sense and make the code more readable?

Comment (etherealjoy) [reply modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/CppPistacheServerCodegen.java]: You are right, it is the correct way. the super class `AbstractCppCodegen` does it correctly already
```C++
    public String toApiName(String type)
```

Updated

**Comment by etherealjoy** [comment]
@stkrwork 
I need to fix the same thing for `cpprest` should I do it here?

**Comment by stkrwork** [comment]
if you are already looking into it, can you also check it in restbed generator?

**Comment by etherealjoy** [comment]
OK I will check it also

**Comment by stkrwork** [comment]
thanks

**Pull request merged**